SwiftCompilerSources/Sources/Optimizer/FunctionPasses/ReleaseDevirtualizer.swift

```diff
@@ -0,0 +1,214 @@
+
+  var object: Value = allocRefInstruction
+  if object.type != type {
+    object = builder.createUncheckedRefCast(object: object, type: type)
+  }
+
+  // Do what a release would do before calling the deallocator: set the object
+  // in deallocating state, which means set the RC_DEALLOCATING_FLAG flag.
+  builder.createSetDeallocating(operand: object, isAtomic: release.isAtomic)
+
+  // Create the call to the destructor with the allocated object as self
+  // argument.
+  let functionRef = builder.createFunctionRef(dealloc)
+
+  let substitutionMap = context.getContextSubstitutionMap(for: type)
+  builder.createApply(function: functionRef, substitutionMap, arguments: [object])
+  context.erase(instruction: release)
+}
+
+private func stripRCIdentityPreservingInsts(_ value: Value) -> Value? {
+  switch value {
+  // First strip off RC identity preserving casts.
+  case let inst as Instruction where inst is UpcastInst ||
+    inst is UncheckedRefCastInst ||
+    inst is InitExistentialRefInst ||
+    inst is OpenExistentialRefInst ||
+    inst is RefToBridgeObjectInst ||
+    inst is BridgeObjectToRefInst ||
+    inst is ConvertFunctionInst ||
+    inst is UncheckedEnumDataInst:
+    return inst.operands[0].value
+
+  // Then if we have a struct_extract that is extracting a non-trivial member
+  // from a struct with no other non-trivial members, a ref count operation on
+  // the struct is equivalent to a ref count operation on the extracted
+  // member. Strip off the extract.
+  case let sei as StructExtractInst where sei.isFieldOnlyNonTrivialField:
+    return sei.operands[0].value
+
+  // If we have a struct instruction with only one non-trivial stored field, the
+  // only reference count that can be modified is the non-trivial field. Return
+  // the non-trivial field.
+  case let si as StructInst:
+    return si.uniqueNonTrivialOperand
+
+  // If we have an enum instruction with a payload, strip off the enum to
+  // expose the enum's payload.
+  case let ei as EnumInst where !ei.operands.isEmpty:
+    return ei.operands[0].value
+
+  // If we have a tuple_extract that is extracting the only non trivial member
+  // of a tuple, a retain_value on the tuple is equivalent to a retain_value on
+  // the extracted value.
+  case let tei as TupleExtractInst where tei.isEltOnlyNonTrivialElt:
+    return tei.operands[0].value
+
+  // If we are forming a tuple and the tuple only has one element with reference
+  // semantics, a retain_value on the tuple is equivalent to a retain value on
+  // the tuple operand.
+  case let ti as TupleInst:
+    return ti.uniqueNonTrivialOperand
+
+  default:
+    return nil
+  }
+}
+
+private extension Instruction {
+  /// Search the operands of this tuple for a unique non-trivial elt. If we find
+  /// it, return it. Otherwise return `nil`.
+  var uniqueNonTrivialOperand: Value? {
+    var candidateElt: Value?
+    let function = self.function
+
+    // For each operand...
```

Comment:
This comment is redundant - you can remove it.
Comments should describe things which are not obvious from the program code and not just do a 1:1 translation from swift syntax to English.